### PR TITLE
Aggressive marketing

### PR DESCRIPTION
**Pull request by Lohikar**
This PR tweaks the 'Brand Intelligence' event to be a bit more meaningful & to punish crews who ignore it. (Based on /tg/'s version)

The initial announcement will now list what type of vendor is infected, and the infected machine will periodically say different advertisement messages than normal. If it is reset, all other hacked machines reset and the event ends. If all vendors on the map become infected, the event gets dangerous. Vendors have a 70% chance to uproot themselves and start chasing people, doing brute melee damage. Uprooted vendors are simple_animals and can be killed normally. A killed vendor returns to normal and can be used again. If a vendor does not uproot itself, it has a 30% chance to explode (fairly weak). Otherwise, nothing happens to the vendor.

Both vendor explosions & vendor uprooting can be set independently via. game config.

As a result of the increased danger of the event, it has been promoted to Moderate severity.

Other changes include grammar tweaks for vendors, tweaks to simple_animal health text, vendors showing speech bubbles, reformatting of the events list, and use of timers in vendors instead of spawn.

### CODE REVIEW

**Comment by LordFowl** [comment]
Why is this any better than APCs randomly exploding? Why is this any better bluespace bears? This seems like a perfect combination of what makes those events bad. Random explosions throughout the round and not insignificant mobs spawning directly inside the station, anywhere.

**Comment by Lohikar** [comment]
This event is easily preventable; locating the origin machine and resetting it will reset other vendors and prevent them going rogue. The announcement states what type of vendor is rogue, and the rogue vendor says abnormal messages every now and then.

**Comment by OneOneThreeEight** [comment]
Hail corporate. +1

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -607,9 +607,15 @@
 	if (!message)
 		return
 
-	for(var/mob/O in hearers(src, null))
-		O.show_message("<span class='game say'><span class='name'>\The [src]</span> beeps, \"[message]\"</span>",2)
-	return
+	var/list/hear = hearers(src, null)
+	var/list/client/clients = list()
+	for (var/mob/M in hear)
```
Comment (skull132) [inline code/game/machinery/vending.dm]: Can't this be refactored into `visible_message()` ? Or nag due to it spoofing normal speaking?

File: code/modules/events/brand_intelligence.dm [hunk 2]
```diff
@@ -6,48 +6,72 @@
...
 	var/obj/machinery/vending/originMachine
 
+	var/list/rampant_speeches = list(
+		"Try our aggressive new marketing strategies!",
+		"You should buy products to feed your lifestyle obession!",
+		"Consume!",
+		"Your money can buy happiness!",
+		"Engage direct marketing!",
+		"Advertising is legalized lying! But don't let that put you off our great deals!",
+		"You don't want to buy anything? Yeah, well I didn't want to buy your mom either."
```
Comment (skull132) [inline code/modules/events/brand_intelligence.dm]: 👌

**Comment by skull132** [comment]
> Uprooted vendors are simple_animals and can be killed normally. A killed vendor returns to normal and can be used again. If a vendor does not uproot itself, it has a 30% chance to explode (fairly weak).

I fear this is too far in /tg/ meme territory for us. Why not give vending machines explody timers or give them a chance to electrocute passersby/users instead, until fixed? It's much more believable and less memey.

**Comment by Lohikar** [comment]
I'm up for removing the explosions from vendors, though I feel like removing their ability to move around would make this event lose a lot of its danger.

**Comment by skull132** [comment]
> I'm up for removing the explosions from vendors, though I feel like removing their ability to move around would make this event lose a lot of its danger.

Well, yes, it's a moderate event. Also, if you make them cause minor explosions after some time uncared for, it'd become a nuisance right quick.

**Comment by Lohikar** [comment]
It's also an easily preventable event however. The vendors only go hostile if they're are completely ignored long enough for all of them to get infected and the origin machine isn't fixed.

File: code/controllers/verbs.dm [hunk 3]
```diff
@@ -9,7 +9,6 @@
 		message_admins("Admin [key_name_admin(usr)] is debugging the [antag.role_text] template.")
 
 /var/list/controller_debug_list = list(
-	"Jobs",
```
Comment (Arrow768) [inline code/controllers/verbs.dm]: Intentional ?
This change is not noted in the PR description.

Comment (Lohikar) [inline code/controllers/verbs.dm]: Intentional, I forgot to remove it when I ported jobs to a subsystem.

**Comment by skull132** [comment]
> It's also an easily preventable event however. The vendors only go hostile if they're are completely ignored long enough for all of them to get infected and the origin machine isn't fixed.

Eh, still. Seems too zany :l

File: code/game/machinery/vending.dm (hunk 1 above)
Comment (Lohikar) [inline code/game/machinery/vending.dm]: It's manually doing it so the speech bubbles work (also equivalent to `audible_message()`), and I'm not sure I want to add them to everything that calls that proc.

**Comment by Lohikar** [comment]
Closing until someone can suggest an alternative that maintains the added danger while still making sense.

**Comment by skull132** [comment]
I did though :(

Stuff shocking passers by and eventually exploding is quite a bit of danger.

**Comment by skull132** [comment]
Or put it up for discussion on suggestion boards. Folks might have ideas.

**Comment by Lohikar** [comment]
Thread created: https://forums.aurorastation.org/viewtopic.php?f=18&t=8214

**Pull request closed without merging**